src/notifications.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright 2015 The Incremental DOM Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Called once a Node (Text or Element) is attached to the DOM.
```

Comment:
I was for `created` and `deleted`. I think it's important to divorce these notifications from DOM manipulations. They aren't "insert", "remove" and "replace" methods, they're just notifications so I can manage the nodes.